### PR TITLE
VZ-10205: Conditionally install verrazzano-ingress-nginx namespace when ingressNGINX is enabled

### PR DESCRIPTION
**Pull request by saidu941**
This PR contains changes for the conditionally install verrazzano-ingress-nginx when ingressNginx is enabled

## Validation
```
[sdosapat@sdosapat-1 verrazzano-scripts]$ time vz install -m https://objectstorage.....sdosapat%2FVZ-10205-2%2Foperator.yaml -s profile=none
2023-07-13T19:10:31.419Z info Finished reconciling Verrazzano resource default/verrazzano

real	0m14.310s
user	0m0.498s
sys	0m0.099s
```

```
[sdosapat@sdosapat-mac verrazzano]$ kubectl get ns -w
NAME                 STATUS   AGE
default              Active   3m8s
kube-node-lease      Active   3m10s
kube-public          Active   3m10s
kube-system          Active   3m10s
local-path-storage   Active   3m4s
metallb-system       Active   3m2s
monitoring           Active   29s
verrazzano-install   Active   39s
verrazzano-system    Active   0s
kube-system          Active   4m1s
verrazzano-mc        Active   0s
cattle-system        Active   0s
verrazzano-monitoring   Active   0s
```

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/common/namespace.go [hunk 1]
```diff
@@ -95,6 +98,15 @@ func CreateAndLabelNamespaces(ctx spi.ComponentContext) error {
 	return nil
 }
 
+func isAnyMonitoringComponentEnabled(ctx spi.ComponentContext) bool {
```
Comment (kminder) [inline platform-operator/controllers/verrazzano/component/common/namespace.go:101]: There needs to be a unit test for this.

Comment (saidu941) [reply platform-operator/controllers/verrazzano/component/common/namespace.go:105]: Addressed and updated PR with UT added

**Review by kminder — APPROVED** [review]
Approved.

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**